app/assets/javascripts/discourse/app/components/user-menu/menu.js

```diff
@@ -1,40 +1,109 @@
 import GlimmerComponent from "discourse/components/glimmer";
 import { tracked } from "@glimmer/tracking";
 import { action } from "@ember/object";
+import { LIKE_NOTIFICATION_FREQUENCY_TYPE } from "discourse/models/user";
+import UserMenuTab from "discourse/lib/user-menu/tab";
 
 const DEFAULT_TAB_ID = "all-notifications";
 const DEFAULT_PANEL_COMPONENT = "user-menu/notifications-list";
 
+const coreTopTabs = [
+  class extends UserMenuTab {
+    get id() {
+      return DEFAULT_TAB_ID;
+    }
+
+    get icon() {
+      return "bell";
+    }
+
+    get panelComponent() {
+      return DEFAULT_PANEL_COMPONENT;
+    }
+  },
+
+  class extends UserMenuTab {
+    get id() {
+      return "replies";
+    }
+
+    get icon() {
+      return "reply";
+    }
+
+    get panelComponent() {
+      return "user-menu/replies-notifications-list";
+    }
+  },
+
+  class extends UserMenuTab {
+    get id() {
+      return "mentions";
+    }
+
+    get icon() {
+      return "at";
+    }
+
+    get panelComponent() {
+      return "user-menu/mentions-notifications-list";
+    }
+  },
+
+  class extends UserMenuTab {
+    get id() {
+      return "likes";
+    }
+
+    get icon() {
+      return "heart";
+    }
+
+    get panelComponent() {
+      return "user-menu/likes-notifications-list";
+    }
+
+    get shouldDisplay() {
+      return (
+        this.currentUser.like_notification_frequency !==
+        LIKE_NOTIFICATION_FREQUENCY_TYPE.never
```

Comment:
Instead of having to do this comparison on the client side, I think a better way will be to have the server side send down an attribute which states whether likes notifications should be displayed for the user.